src/Microsoft.OData.Core/Json/ODataUtf8JsonWriter.Stream.cs

```diff
@@ -0,0 +1,190 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="ODataUtf8JsonWriter.Stream.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+#if NETCOREAPP

```

Comment:
This seems like a good opportunity to introduce proper polymorphism. These if based branches can only take us so far.

Reply:
@wachugamaina  This feature is not supported in the .net framework.. That's why you're seeing these ifs.. I'd like to know though what you had in mind on how this can be done.